atom/browser/api/atom_api_url_request.cc

```diff
@@ -0,0 +1,462 @@
+// Copyright (c) 2013 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include <string>
+#include "atom/browser/api/atom_api_session.h"
+#include "atom/browser/api/atom_api_url_request.h"
+#include "atom/browser/net/atom_url_request.h"
+#include "atom/common/native_mate_converters/callback.h"
+#include "atom/common/native_mate_converters/net_converter.h"
+#include "atom/common/native_mate_converters/string16_converter.h"
+#include "atom/common/node_includes.h"
+#include "native_mate/dictionary.h"
+
+
+
+namespace mate {
+
+template<>
+struct Converter<scoped_refptr<const net::HttpResponseHeaders>> {
+  static v8::Local<v8::Value> ToV8(
+    v8::Isolate* isolate,
+    scoped_refptr<const net::HttpResponseHeaders> val) {
+    mate::Dictionary dict = mate::Dictionary::CreateEmpty(isolate);
+    if (val) {
+      size_t iter = 0;
+      std::string name;
+      std::string value;
+      while (val->EnumerateHeaderLines(&iter, &name, &value)) {
```

Comment:
Conversion should mimic https://github.com/electron/electron/blob/master/atom/browser/api/atom_api_web_contents.cc#L146 , maybe converter can be moved to `atom/commom/native_mate_converters/net_converter`